### ISSUE TITLE
TeamCity Compatability Question?

### ISSUE THREAD

**Issue by chumphreyTMW** [rossrowe/sauce-ci-java-demo#3]
Please can you confirm if the tutorial here on  the link below is compatible with TeamCity 9.0.4. 32407 ?

https://docs.saucelabs.com/ci-integrations/teamcity/#teamcity-configuration-for-a-java-based-project

**Comment by rossrowe** [comment]
Hi, I haven't tried TeamCity 9.0.4 yet, but I'll download it and give it a shot, however I'd expect the project to still work with the latest version of TeamCity.  If you have any specific errors when running a build with the demo project, please let me know

**Comment by chumphreyTMW** [comment]
Cheers Ross.  When you get the time take a look at this.

Regards,

C

**Comment by rossrowe** [comment]
Hi again, just tried TeamCity 9.0.4, and the sample project seemed to run
okay for me, please let me know if you get any errors running it in your
environment

**Comment by chumphreyTMW** [comment]
Hi Ross,

Thanks for running your tests against the newest build 9.0.4.

Myself and some of the guys from Magentys did some debugging and noticed
that it appears if SAUCE_ONDEMAND_BROWSERS is not selected from browser matrix
then a NullPointerException is thrown because the browser array is empty.

Also, have you got any  documentation about the Sauce Labs Results Tab?
I gather perhaps this tab should output some information when tests are run
against sauce.

I confirm tests are run by logging into SauceLabs however nothing is generated in
I can assist with any form of logs you may require or even team city project backups.

I look forward to hearing from you soon.

Kind regards,

Colin

From: Marcus Petrat <marcus.petrat@magentys.io<mailto:marcus.petrat@magentys.io>>
Date: Monday, 8 June 2015 17:46
To: Colin Humphrey <CHumphrey@tmwunlimited.com<mailto:CHumphrey@tmwunlimited.com>>
Cc: Hamish Tedeschi <hamish.tedeschi@magentys.io<mailto:hamish.tedeschi@magentys.io>>, Rolff Kruger <RKruger@tmwunlimited.com<mailto:RKruger@tmwunlimited.com>>, Iqbal Khan <IKhan@tmwunlimited.com<mailto:IKhan@tmwunlimited.com>>
Subject: Re: Further to the consultation 12pm today

Hello Colin,

thank you for sending me the error log. As with regards to this error in the log:

1) To the first error in the log:

15:59:36] : [com.saucelabs:sauce-ci-demo] java.lang.NullPointerException
[15:59:36] : [com.saucelabs:sauce-ci-demo] at java.io.StringReader.<init>(StringReader.java:50)
[15:59:36] : [com.saucelabs:sauce-ci-demo] at org.json.JSONTokener.<init>(JSONTokener.java:66)
[15:59:36] : [com.saucelabs:sauce-ci-demo] at org.json.JSONArray.<init>(JSONArray.java:159)
[15:59:36] : [com.saucelabs:sauce-ci-demo] at com.saucelabs.junit.WebDriverWithHelperJSONTest.data(WebDriverWithHelperJSONTest.java:117)

I have had a look at this class in the actual github repository (https://github.com/rossrowe/sauce-ci-java-demo).

This is the code where the failure is:

/**
        \* JUnit annotation that runs each test once for each item in a Collection.
        *
        \* Feel free to add as many additional parameters as you like to the capabilitiesParams array.
        *
        \* Note: If you add parameters for the MAC platform, make sure that you have Mac minutes in
        \* your <a href="https://saucelabs.com/login">Sauce account</a> or the test will fail.
        */
        @Parameters
        public static Collection<Object[]>data() {

```
    String json =System.getenv("SAUCE_ONDEMAND_BROWSERS");
    if (json == null) {
    json = System.getenv("bamboo_SAUCE_ONDEMAND_BROWSERS");
    }
    List<Object[]> browsers = new ArrayList<Object[]>();
    JSONArray browserArray = null;
    try {
    browserArray = new  JSONArray(json); -> failure! ... It loooks like the json string is not being populated. Have you set the variable SAUCE_ONDEMAND_BROWSERS ?Also, in the comments it mentions some necessary minutes to be added for the Mac platform in Saucelabs (don't think that you have configured the tests to run on the Mac platform)  2) With this error: Unrecognized Selenium driver URI 'embedded-rc:'. Make sure you got the proper driver jars in your classpath, or increase the logging level to get more information.
```

Have you got the driver jars in your classpath? This should be the 'embedded Selenium RC driver' as mentioned here: http://selenium-client-factory.infradna.com/driver-embeddedRC.html Kind regards, Marcus

On 8 June 2015 at 14:34, Colin Humphrey <CHumphrey@tmwunlimited.com<mailto:CHumphrey@tmwunlimited.com>> wrote:
Hi Guys,

Further to my previous email.

1, Verify presence of Magentys/TAF TMW Teamcity Server and if its cloneable.
    Verified the presence of several TeamCity internal build projects.  Investigating cloning options.

2, Of the two links below sent by Hamish.
https://www.jetbrains.com/teamcity/features/maven.html // First link not so useful
https://docs.saucelabs.com/ci-integrations/teamcity/#teamcity-configuration-for-a-java-based-project
// The build works with mixed failing/passing tests. Have emailed original repository author with error.log.  Same attached here.  Feedback awaited.
  Have successfully completed the sauce labs canonical maven/java tutorial

3, Create TeamCity project and Wiki write up
  In progress now

Thanks,

Colin

From: Colin Humphrey <CHumphrey@tmwunlimited.com<mailto:CHumphrey@tmwunlimited.com>>
Date: Monday, 8 June 2015 13:03
To: Marcus Petrat <marcus.petrat@magentys.io<mailto:marcus.petrat@magentys.io>>
Cc: Hamish Tedeschi <hamish.tedeschi@magentys.io<mailto:hamish.tedeschi@magentys.io>>, Rolff Kruger <RKruger@tmwunlimited.com<mailto:RKruger@tmwunlimited.com>>, Iqbal Khan <IKhan@tmwunlimited.com<mailto:IKhan@tmwunlimited.com>>
Subject: Further to the consultation 12pm today

Hi Marcus,

Thank you for your time before.  As discussed there are several steps to try with TeamCity integration / Magentys Maven Project.
The activity will time boxed to today at which point progressed will be reviewed and on site consultation - as suggested - will be considered.

1, Verify presence of Magentys/TAF TMW Teamcity Server and if its cloneable.

2, Of the two links below sent by Hamish.
https://www.jetbrains.com/teamcity/features/maven.html // First link not so useful
https://docs.saucelabs.com/ci-integrations/teamcity/#teamcity-configuration-for-a-java-based-project
// The build works with mixed failing/passing tests. Have emailed original repository author with error.log.  Same attached here.  Feedback awaited.

3, Create TeamCity project

Colin Humphrey
Senior PHP Developer

[TMW Unlimited]

T +44 (0)20 7751 5447tel:%2B44%20%280%2920%207751%205447 | M
Creston House, 10 Great Pulteney Street, London, W1F 9NB
CHumphrey@tmwunlimited.commailto:CHumphrey@tmwunlimited.com | www.tmwunlimited.comhttp://www.tmwunlimited.com
A part of Creston Unlimitedwww.creston-unlimited.comhttp://www.creston-unlimited.com

---

As part of the next phase of our development, we are now TMW Unlimited with a broader range of services and an even more sophisticated offering.
We are also a part of a new group brand offer called Creston Unlimited which brings together discipline experts from 25 specialist agencies within our group to provide thought leadership, capabilities and tools to help clients navigate through today's ever-changing landscape.
Welcome to a world of thinking without limits.
To find out more about TMW Unlimited and our new group brand offer please contact Richard Marshall at rmarshall@tmwunlimited.commailto:rmarshall@tmwunlimited.com

---

Database Marketing Award
Winner of the Innovation in Insight-Driven Social Media Marketing for Lynx Brand Love: The Rebound.

DMA Awards
Winner of 3 Golds in the FMCG category for Lynx Peace and in the Pharmaceutical and Best Brand Building categories for Durex #TurnOffToTurnOn

Webby Awards Nominee
Official Nominee in the Interactive, Wellness, Health and Pharma category with Durex #TurnOffToTurnOn.

This email is intended only for the person to whom it is addressed. If an addressing or transmission error has misdirected this email, please notify the sender by replying to this email. If you are not the intended recipient, please delete this email and do not use, disclose, copy, print or rely on the email in any manner. To the extent permitted by law, Tullo Marshall Warren Limited does not accept or assume any liability, responsibility or duty of care for any use of or reliance on this email by anyone, other than the intended recipient to the extent agreed in the relevant contract for the matter to which this email relates (if any).
Tullo Marshall Warren Limited is a limited liability company registered in England with registered number 01332638. The registered office of Creston plc is 10 Great Pulteney Street, London W1F 9NB.

**Comment by rossrowe** [comment]
Hi Colin, yes if a single browser is selected, then the SAUCE_ONDEMAND_BROWSERS environment variable doesn't currently get populated, only the SELENIUM_\* environment variable get populated.  I've updates the plugin to ensure that the SAUCE_ONDEMAND_BROWSERS environment variable always gets populated, which is available in version 1.32 (available from https://repository-saucelabs.forge.cloudbees.com/release/com/saucelabs/teamcity/sauceplugin/1.32/sauceplugin-1.32.zip).

With regards to the result tab, the plugin will display the Sauce jobs that have been associated with the TeamCity build.  This association can be performed by including the following output in the build output of your build:

SauceOnDemandSessionID=SESSION_ID job-name=JOB_NAME

There's some further information available on how to extract the session id contained at https://docs.saucelabs.com/ci-integrations/teamcity/#output-sauce-session-id

As of version 1.32, you can also link the job to the Team City build by setting the 'build' capability (using the value of the SAUCE_TC_BUILDNUMBER environment variable)

**Comment by chumphreyTMW** [comment]
Thanks Ross,

That's great advice.  I'll look at that shortly.
I think the plugin is a great way to use Sauce in TeamCity.  Fairly new to them both so thanks again for the assist.

Have a great day.

Regards,

C

**Comment by chumphreyTMW** [comment]
Hi Again,

I thought I’d quickly get in touch and confirm that in principle I understand what needs to be done however
I don’t have enough familiarity to add these steps in the TeamCity.

If at all possible could you be a little more step driven in outlining how to get the values into the build output.
Not sure where the Build Output is or how it exists in team city as a concept.

Thanks again and kind regards,

Colin

**Comment by rossrowe** [comment]
There's a tutorial on using the TeamCity plugin at https://docs.saucelabs.com/ci-integrations/teamcity, but basically the plugin will look at the output of your build (ie, the text that's written to the build log) and try to find output lines with the 'SauceOnDemandSessionID' format, and for each matching line, will invoke the Sauce REST API to store the team city build number against the Sauce job.

If you're using Java, then the easiest way to output the line is to write the output via System.out.println() as per https://docs.saucelabs.com/ci-integrations/teamcity/#output-sauce-session-id.

You should also be able to associate the TeamCity build number with the Sauce job by adding the following code to your DesiredCapabilities creation (again using Java as an example):

```
capabilities.setCapability("build", System.getenv("SAUCE_TC_BUILDNUMBER"));
```

However, if you're using the sample code that's included with this project, then the SauceOnDemandTestWatcher variable (which is included in SauceOnDemandTest) should already be including the correct output to the stdout - can you let me know if your tests are using the SauceOnDemandTestWatcher, and if so, whether the SauceOnDemandSessionID is included in your build output?

**Comment by chumphreyTMW** [comment]
Hi Ross,

I just noticed this email was sitting in drafts and did not get sent.  See my responses inline in orange below.

Thanks,

Colin

**Comment by rossrowe** [comment]
Okay, in that case it's probably worth raising a support request with Sauce Labs (requesting that the ticket get assigned to me), so that I can take a closer look at the Sauce jobs which were run.  At least in my environment, when running this sample project with TeamCity 9.0.4, the Sauce Labs Results tab is correctly populated with the tests which were run, without me having to make any modifications to the test code.

**Comment by chumphreyTMW** [comment]
Again Ross,

Thanks for the confirmation.  That's good news.
I'll take a closer look in parallel to your efforts.

There are several options for me, get the same project installed on our internal
server.

I'll raise the request shortly.

Kind regards,

C

**Comment by chumphreyTMW** [comment]
Hi Ross,

I have created the support ticket.  Just to be absolutely sure.  You ran the packaged/zipped up code
which attached and that correctly populated the SauceLabs Results Tab.

I just want to make sure that it was not respoitory / sauce-ci-java-demo which you ran.  I do not want to
prematurely communicate any errata.

Thanks,

C

**Comment by rossrowe** [comment]
No, I ran the sample project against my TeamCity instance, I haven't received any attachments (I don't think Github handles adding attachments to issues) - if you attach your sample files to the support issue, then we can take a look at your code to see where the issue might be

**Comment by chumphreyTMW** [comment]
I will do that.  Were you able to view the screen casts.  I am also able to successfully get data back from your sample project
however screen cast videos do not load.

I’ll attach to the ticket as soon as possible.

Thanks,

C

**Comment by rossrowe** [comment]
Yes, I just checked the screencast display in my TeamCity environment, and
I am also having issues getting it to play - it's probably worth adding
this issue into the ticket as well (it doesn't look like it's related to
the plugin specifically, as the embedded report is displaying some
information correctly)

**Comment by chumphreyTMW** [comment]
## Resent as first email failed on account of file size, will send link to cload based file share shortly.

Hi Ross,

Thanks for the support once more.

Using the repo here: https://github.com/rossrowe/sauce-ci-java-demo/

Please find below the answers to your questions.

1 Am I using SauceOnDemandTestWatcher:
Yes the symbol is being used in the test  sauce-ci-java-demohttps://github.com/rossrowe/sauce-ci-java-demo/srchttps://github.com/rossrowe/sauce-ci-java-demo/tree/master/src/testhttps://github.com/rossrowe/sauce-ci-java-demo/tree/master/src/test/javahttps://github.com/rossrowe/sauce-ci-java-demo/tree/master/src/test/java/SauceOnDemandTest.java

2, Is the SauceOnDemandSessionID output in the logs.
Yes it is.

In a nutshell the SauceLabs Results Tab is working.  However video does not appear to play.
My issue is with integrating the Sauce Labs Results Reporting Tab into a personal repository.
See the zip attached.  I will see if there is a way of learning from the your tutorial.

Regards,

C

**Comment by chumphreyTMW** [comment]
As before, due to file size limits recent emails have not been gettinh through.
Resent as first email failed on account of file size, will send link to cload based file share shortly.

Hi Ross,

Thank you for the feedback, just checking do you want these raised via github or your
Company site?  I can confirm I have some basic wiring between sauce labs and my local team city
builds.

Test session id data and a job-name are output,
The session id is output but the name is returned null, perhaps its in the wrong format.
Attached is the latest codebase, log and screen capture of sauce labs reporting tab showing tests populated.

CURL COMMAND

curl -u magentys_tmw:cc2d5e56-a062-4a1c-a6f1-f143f46f2eff \
https://saucelabs.com/rest/v1/magentys_tmw/jobs?limit=10

[{"id": "ba319817f84e478ab53835490c50df69"}, {"id": "3ddee692b785456983b67bed20de6d35"}, {"id": "38f51d72778f43ec94a55ccf38ed1643"}, {"id": "d0b9577f8e9743748618605347ad81ba"}, {"id": "89eac82b463d4451b2d99605e8856b84"}, {"id": "52e3a58965b64a88854c266840db335d"}, {"id": "d1042e5fda4d48c4913d67675ed1b351"}, {"id": "512cb5894ba249ebbc5a808a4cbcae54"}, {"id": "480970a634384fbcade93d4f476b8855"}, {"id": "b0d69abebb834a25b9de4e76308b3308"}]chumphrey:diageo-the-bar chumphrey$

SCREENSHOT

There does appear to  be a correlate between the recent session ids.  These appear from the CURL command as well as the screen capture.  Please see below.

89eac82b463d4451b2d99605e8856b84
d0b9577f8e9743748618605347ad81ba

LOG OUTPUT
Below is a sample from my log as well.
[11:23:27][com.tmw:sample-java-automation-framework] SauceOnDemandSessionID=89eac82b463d4451b2d99605e8856b84 job-name=org.junit.rules.TestName@5438c785
[11:23:33][com.tmw:sample-java-automation-framework]
[11:23:33][com.tmw:sample-java-automation-framework] Background: [90m# example.feature:7 [0m
[11:23:33][com.tmw:sample-java-automation-framework] [32mGiven [0m [32mI have visited the example url [0m [90m# ExampleStepDefs.i_have_visited_the_example_url() [0m
[11:25:05][com.tmw:sample-java-automation-framework] SauceOnDemandSessionID=d0b9577f8e9743748618605347ad81ba job-name=org.junit.rules.TestName@1837acad
[11:26:23][com.tmw:sample-java-automation-framework] [31mFailure in after hook: [0m [31mSauceWebDriverProvider.quit() [0m
[11:26:23][com.tmw:sample-java-automation-framework] [31mMessage: [0m [31morg.openqa.selenium.UnsupportedCommandException: ERROR The test with session id 52e3a58965b64a88854c266840db335d has already finished, and can't receive further commands
[11:26:23][com.tmw:sample-java-automation-framework] You can learn more at https://saucelabs.com/jobs/52e3a58965b64a88854c266840db335d
[11:26:23][com.tmw:sample-java-automation-framework] For help, please check https://docs.saucelabs.com/reference/troubleshooting-common-error-messages

Kind regards,

C

**Comment by chumphreyTMW** [comment]
Please can you confirm if this is the correct email for yourself?
  piaross@gmail.com

From: Colin Humphrey <CHumphrey@tmwunlimited.com<mailto:CHumphrey@tmwunlimited.com>>
Date: Thursday, 11 June 2015 12:14
To: rossrowe/sauce-ci-java-demo <reply@reply.github.com<mailto:reply@reply.github.com>>
Cc: Colin Humphrey <CHumphrey@tmwunlimited.com<mailto:CHumphrey@tmwunlimited.com>>
Subject: Re: [sauce-ci-java-demo] TeamCity Compatability Question? (#3)

As before, due to file size limits recent emails have not been gettinh through.
Resent as first email failed on account of file size, will send link to cload based file share shortly.

Hi Ross,

Thank you for the feedback, just checking do you want these raised via github or your
Company site?  I can confirm I have some basic wiring between sauce labs and my local team city
builds.

Test session id data and a job-name are output,
The session id is output but the name is returned null, perhaps its in the wrong format.
Attached is the latest codebase, log and screen capture of sauce labs reporting tab showing tests populated.

CURL COMMAND

curl -u magentys_tmw:cc2d5e56-a062-4a1c-a6f1-f143f46f2eff \https://saucelabs.com/rest/v1/magentys_tmw/jobs?limit=10

[{"id": "ba319817f84e478ab53835490c50df69"}, {"id": "3ddee692b785456983b67bed20de6d35"}, {"id": "38f51d72778f43ec94a55ccf38ed1643"}, {"id": "d0b9577f8e9743748618605347ad81ba"}, {"id": "89eac82b463d4451b2d99605e8856b84"}, {"id": "52e3a58965b64a88854c266840db335d"}, {"id": "d1042e5fda4d48c4913d67675ed1b351"}, {"id": "512cb5894ba249ebbc5a808a4cbcae54"}, {"id": "480970a634384fbcade93d4f476b8855"}, {"id": "b0d69abebb834a25b9de4e76308b3308"}]chumphrey:diageo-the-bar chumphrey$

SCREENSHOT

There does appear to  be a correlate between the recent session ids.  These appear from the CURL command as well as the screen capture.  Please see below.

89eac82b463d4451b2d99605e8856b84
d0b9577f8e9743748618605347ad81ba

LOG OUTPUT
Below is a sample from my log as well.
[11:23:27][com.tmw:sample-java-automation-framework] SauceOnDemandSessionID=89eac82b463d4451b2d99605e8856b84 job-name=org.junit.rules.TestName@5438c785
[11:23:33][com.tmw:sample-java-automation-framework]
[11:23:33][com.tmw:sample-java-automation-framework] Background: [90m# example.feature:7 [0m
[11:23:33][com.tmw:sample-java-automation-framework] [32mGiven [0m [32mI have visited the example url [0m [90m# ExampleStepDefs.i_have_visited_the_example_url() [0m
[11:25:05][com.tmw:sample-java-automation-framework] SauceOnDemandSessionID=d0b9577f8e9743748618605347ad81ba job-name=org.junit.rules.TestName@1837acad
[11:26:23][com.tmw:sample-java-automation-framework] [31mFailure in after hook: [0m [31mSauceWebDriverProvider.quit() [0m
[11:26:23][com.tmw:sample-java-automation-framework] [31mMessage: [0m [31morg.openqa.selenium.UnsupportedCommandException: ERROR The test with session id 52e3a58965b64a88854c266840db335d has already finished, and can't receive further commands
[11:26:23][com.tmw:sample-java-automation-framework] You can learn more at https://saucelabs.com/jobs/52e3a58965b64a88854c266840db335d
[11:26:23][com.tmw:sample-java-automation-framework] For help, please check https://docs.saucelabs.com/reference/troubleshooting-common-error-messages

Kind regards,

C

**Comment by rossrowe** [comment]
Yes, please raise a support ticket for the screencast issue through support.saucelabs.com.

**Comment by chumphreyTMW** [comment]
Thanks Ross,

Done.

Please also confirm your personal email.  I’d like to send you a link to a resource containing the latest
Sample-project, screen captures and logs. I have coded this as an example to integrate with TC / Magentys and SauceLabs plugin.

Kind regards,

Colin

**Comment by rossrowe** [comment]
Yep, email address is correct, although it's probably best if you include
that info in the support ticket too

**Comment by saucyallison** [comment]
Hello from Sauce Labs Support, @chumphreyTMW! The screencast issue in TeamCity should now be fixed. If you could confirm this, we'd greatly appreciate it. Thanks a bunch!
